frontend/lib/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -243,6 +243,20 @@ const RawElementNodeRenderer = (
   // since leaf elements are always direct children of a VerticalBlock, which always calculates
   const width = props.width ?? 0
 
+  if (
+    node.element.type &&
+    props.componentMapper &&
+    props.componentMapper[node.element.type]
+  ) {
+    const componentFunc = props.componentMapper[node.element.type]
+    switch (node.element.type) {
+      case "heading":
+        return componentFunc({
+          value: node.element.heading?.body,
```

Comment:
here's the transform from protobuf element to the properties to pass to the custom "heading" component